src/main/java/co/novu/api/subscribers/SubscribersHandler.java

```diff
@@ -31,80 +35,84 @@ public class SubscribersHandler {
-        return restHandler.handlePost(request, CreateBulkSubscriberResponse.class, novuConfig, ENDPOINT + "/bulk");
+    public CreateBulkSubscriberResponse createSubscriberBulk(BulkSubscriberRequest request) throws IOException, NovuNetworkException {
+        return restHandler.extractResponse(subscribersApi.createBulkSubscriber(request).execute());
     }
 
-    public SingleSubscriberResponse getSubscriber(String subscriberId) {
-        return restHandler.handleGet(SingleSubscriberResponse.class, novuConfig, ENDPOINT + "/" + subscriberId);
+    public SingleSubscriberResponse getSubscriber(String subscriberId) throws IOException, NovuNetworkException {
+        return restHandler.extractResponse(subscribersApi.getSubscriber(subscriberId).execute());
     }
 
-    public SingleSubscriberResponse updateSubscriber(UpdateSubscriberRequest request, String subscriberId) {
-        return restHandler.handlePut(request, SingleSubscriberResponse.class, novuConfig, ENDPOINT + "/" + subscriberId);
+    public SingleSubscriberResponse updateSubscriber(UpdateSubscriberRequest request, String subscriberId) throws IOException, NovuNetworkException {
+        return restHandler.extractResponse(subscribersApi.updateSubscriber(request, subscriberId).execute());
     }
 
-    public SubscriberDeleteResponse deleteSubscriber(String subscriberId) {
-        return restHandler.handleDelete(SubscriberDeleteResponse.class, novuConfig, ENDPOINT + "/" + subscriberId);
+    public SubscriberDeleteResponse deleteSubscriber(String subscriberId) throws IOException, NovuNetworkException {
+        return restHandler.extractResponse(subscribersApi.deleteSubscriber(subscriberId).execute());
     }
 
-    public SingleSubscriberResponse updateSubscriberCredentials(UpdateSubscriberCredentialsRequest request, String subscriberId) {
-        return restHandler.handlePut(request, SingleSubscriberResponse.class, novuConfig, ENDPOINT + "/" + subscriberId + "/credentials");
+    public SingleSubscriberResponse updateSubscriberCredentials(UpdateSubscriberCredentialsRequest request, String subscriberId) throws IOException, NovuNetworkException {
+        return restHandler.extractResponse(subscribersApi.updateSubscriberCredentials(request, subscriberId).execute());
     }
 
-    public DeleteCredentialsResponse deleteSubscriberCredentials(String subscriberId, String providerId) {
-        boolean isSuccess = restHandler.handleDeleteForVoid(novuConfig, ENDPOINT + "/" + subscriberId + "/credentials/" + providerId);
-        if (isSuccess) {
+    public DeleteCredentialsResponse deleteSubscriberCredentials(String subscriberId, String providerId) throws IOException {
+        Response<Void> deleteSubscriberCredentialsResponse = subscribersApi.deleteSubscriberCredentials(subscriberId, providerId).execute();
+        if (deleteSubscriberCredentialsResponse.isSuccessful()) {
             return new DeleteCredentialsResponse();
+        } else {
+            return null;
         }
```

Comment:
You don't need to check if it's successful, there is a second **_extractResponse()_** function in RestHandler, please use that

Reply:
Hey @mayorJAY the `extractResponse` throws an exception in case the response isn't successful. 

For this method, the existing behavior returns a null for non successful responses. This is why I've implemented it manually. 

Let me know if we're changing the existing behavior.

Reply:
You can have a look at other merged PRs for the expected behaviour

Reply:
Updated the behavior in the latest commit.